core/trino-main/src/main/java/io/trino/operator/GroupByIdBlock.java

```diff
@@ -47,91 +92,109 @@ public long getGroupCount()
 
     public long getGroupId(int position)
     {
-        return BIGINT.getLong(block, position);
+        checkReadablePosition(position);
+        return groups[position & mask];
+    }
+
+    @VisibleForTesting
+    boolean isRle()
+    {
+        return mask == MASK_RLE;
     }
 
     @Override
-    public Block getRegion(int positionOffset, int length)
+    public GroupByIdBlock getRegion(int positionOffset, int length)
     {
-        return block.getRegion(positionOffset, length);
+        // This implementation cannot return region without copying
+        return copyRegion(positionOffset, length);
     }
 
     @Override
     public long getRegionSizeInBytes(int positionOffset, int length)
     {
-        return block.getRegionSizeInBytes(positionOffset, length);
+        return isRle() ? SIZE_OF_ELEMENT : length * SIZE_OF_ELEMENT;
     }
 
     @Override
     public OptionalInt fixedSizeInBytesPerPosition()
     {
-        return block.fixedSizeInBytesPerPosition();
+        return OptionalInt.of(SIZE_OF_ELEMENT);
     }
 
     @Override
     public long getPositionsSizeInBytes(boolean[] positions, int selectedPositionCount)
```

Comment:
this seems inconsistent with `io.trino.spi.block.RunLengthEncodedBlock#getPositionsSizeInBytes`